app/javascript/controllers/qrscanner_controller.js

```diff
@@ -23,8 +23,14 @@ export default class extends Controller {
       return;
     }
     const itemId = splitResult[1];
-    this.close();
-    window.location.href = "/items/" + itemId;
+    
+    fetch("items/" + itemId)
+      .then(response => {
+        if(response.status != 404) {
```

Comment:
```suggestion
        if(response.ok) {
```
I personally would check for a successful status code instead of a specific error.

Reply:
We only check for the status 404 so that other codes like 5xx are still shown.